backend/src/main/java/com/htecgroup/skynest/filter/CustomAuthorizationFilter.java

```diff
@@ -80,11 +82,20 @@ protected void doFilterInternal(
       ExceptionUtil.writeToResponse(errorMessage, response);
     } catch (Exception ex) {
       log.error(ex);
+
+      List<String> messages = new ArrayList<>();
+      Throwable currentException = ex;
+      while (currentException != null && currentException != currentException.getCause()) {
```

Comment:
Can you tell me what is the idea here?

Reply:
you are trying to get the first error in this nested loop ?

Reply:
The earlier version of this wouldn't show errors coming from any of the triggers, since those errors are a cause of a cause of a cause...

Now, users get the whole chain of causes.